### ISSUE TITLE
Run warmup for each chain

### ISSUE THREAD

**Issue by rlouf** [rlouf/mcx#21]
As per discussion in issue #19, it is likely that chains starting at very different positions in the state space will not sample efficiently with the same kernel parameters. It would thus make sense to run one warmup per chain when we warmup.

The sampler would only require a very light modification:

```python
@partial(jax.jit, static_argnums=(1,))
def move_chain(rng_key, kernel, state):
     new_state = kernel(rng_key, state)
     return new_state

jax.vmap(move_chain)(keys, kernels, states)
```

And we will need to vmap the warmup function onto the initial states and parameters.

I am opening this issue to keep track of the idea and welcome discussions until completion.

**Comment by rlouf** [comment]
We have a working prototype in #29.
